src/backend/distributed/transaction/transaction_management.c

```diff
@@ -251,12 +268,11 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 			}
 
 			/*
-			 * Call other parts of citus that need to integrate into
-			 * transaction management. Call them *after* committing/preparing
-			 * the remote transactions, to allow marking shards as invalid
-			 * (e.g. if the remote commit failed).
+			 *
+			 * Check again whether shards/placement successfully
+			 * committed. This handles failure at COMMIT/PREPARE time.
 			 */
-			RouterExecutorPreCommitCheck();
+			CheckForFailedPlacements(false, using2PC);
```

Comment:
Would it make sense to just check the GUC in CheckShardPlacements? I wasn't quite sure what using2PC was referring to.